coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.coral.coralservice.controller;
+
+import static com.linkedin.coral.coralservice.utils.CoralUtils.*;
+import static com.linkedin.coral.coralservice.utils.TranslationUtils.*;
+
+
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * Class defining the REST endpoints to translate queries using Coral.
+ */
+@RestController
+@Service
+@Profile({"hiveMetastoreClient", "default"})
+public class TranslationController implements ApplicationListener<ContextRefreshedEvent> {
+
+  @Override
+  public void onApplicationEvent(ContextRefreshedEvent event) {
+    // runs after the Spring context has been initialized
+    try {
+      initHiveMetastoreClient();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @GetMapping("/translate")
+  public ResponseEntity translate(@RequestParam String query, @RequestParam String fromLanguage, @RequestParam String toLanguage) {
```

Comment:
It is better to annotate these parameters with @nonnull to proactively prevent null value passing into the function body.

Reply:
Springboot makes RequestParams required by default!

I tested it out to make sure so if I send in a request with a param missing I get:
```
{
    "timestamp": "2022-04-14T19:42:45.115+0000",
    "status": 400,
    "error": "Bad Request",
    "message": "Required String parameter 'toLanguage' is not present",
    "path": "/translate"
}
```